src/algorithm/convexhull.rs

```diff
@@ -0,0 +1,174 @@
+use std::cmp::Ordering;
+use num_traits::Float;
+use types::{Point, Polygon, LineString};
+use std::collections::BTreeSet;
+
+impl<T> Eq for Point<T> where T: Float {}
```

Comment:
This is a problem related to punting on `NaN` values – BTreeSet needs `Ord`, but as you say, there's no total ordering for types which included `NaN`. However, I'm not sure that it should be possible to construct geometries with NaN values in the first place.
I feel like there should be a more elegant solution here, but it'll need input from someone who knows more than I do about floating point…